src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,42 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
+
+    public RegistrationServiceImpl(StorageDaoImpl storage) {
+        this.storage = storage;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new RegistrationException("Login and password cannot be null");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters.");
```

Comment:
The checklist specifies to avoid adding something that may change over time, like max user age. However, it does not mention a similar constraint for login length. If the login length requirement is not part of the task, it should not be included. Additionally, ensure that all hardcoded values are moved to constant fields with informative names.